integration/cluster_proxy.go

```diff
@@ -43,7 +52,9 @@ func newClientV3(cfg clientv3.Config) (*clientv3.Client, error) {
 		return nil, err
 	}
 
+	pmu.Lock()
 	proxies[c] = toGRPC(c)
```

Comment:
this assignment should probably happen in `toGRPC`. I think this deadlocks right now?